src/cmds/fs/uniq.c

```diff
@@ -0,0 +1,178 @@
+/**
+ * @file
+ * @brief Filter  adjacent matching lines from FILE, writing to standard output.
+ *
+ * @date 02.07.19
+ * @author Nastya Nizharadze
+ */
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <ctype.h>
+#include <unistd.h>
+#include <errno.h>
+
+#define NL_AND_NUL	2
+
+#define FLAG_DUPLICATES 1 << 4
+#define FLAG_C 1 << 3
+#define FLAG_d 1 << 2
+#define FLAG_D 1 << 1
+#define FLAG_Z 1 << 0
+#define NO_FLAG 0
+
+static void print_usage(void) {
+	printf(
+		"Usage: uniq [OPTION]... [FILE]...\n"
+		"Filter  adjacent matching lines from FILE, "
+		"writing to standard output.\n"
+		"\nOptions:\n\n"
+		"  -c\tprefix lines by the number of occurrences\n"
+		"  -d\tonly print duplicate lines, one for each group\n"
+		"  -D\tprint all duplicate lines\n"
+		"  -z\tline delimiter is NUL, not newline\n"
+		"  -h\tdisplay this help and exit\n"
+	);
+}
+
+static int read_line(char *line_buff, int max_length, int flags, FILE *fp) {
+	char *res;
+	res = fgets(line_buff, max_length, fp);
+	if (!res) {
```

Comment:
You don't use `res` anyway, so you can write
```
if (!fgets(line_buff, max_length, fp)) {
    return -1;
}
```

Reply:
Done.